### PR TITLE
Move npm rules to separate packages

### PR DESCRIPTION
**Pull request by vladmos**
Currently npm rules for buildifier and buildozer are defined in the same packages as the binaries, which means that users who want to use buildifier or buildozer in their bazel builds must depend on `rules_nodejs` (either implicitly by calling `buildifier_dependencies()` from their `WORKSPACE` file or explicitly by defining an external repository).

This change moves the npm targets to a separate `npm` subpackages, which means that users who don't need them should not add an additional dependency.

This change reverts #797 and fixes #787

cc @alexeagle @sudoforge

### CODE REVIEW

**Comment by alexeagle** [comment]
Cool, it's certainly a bug if downstream users observed a rules_nodejs dependency. It's a development-only dependency.

I think the first concern is that my PR adding rules_nodejs should have failed. I think this repo needs some integration testing that looks the way a user would consume it. Then my PR would have been red because the example usage would have required the added repo, which would have been our trigger to consider whether we were breaking the public API.

We have such tests in rules_nodejs so I can give some hints about how to add them if you like - it would help avoid tripping over this again when adding another development-only dependency.

(Greg and I gave a talk about it at bazelcon https://www.youtube.com/watch?v=2KzOLddhdoI)

Moving these to a subpackage seems fine in this case. Another solution is to use fencing like
```
# BEGIN-INTERNAL
codeowners(name = "owners")
# END-INTERNAL
```
which can be required in some cases where the rule has to live in the same package with some sources, but should not be included in the distro installed by users. We have some logic in our packaging rules to strip such regions when building a release artifact.

Thanks for fixing this!

File: WORKSPACE [hunk 1]
```diff
@@ -70,6 +70,13 @@ load("@com_github_bazelbuild_buildtools//buildifier:deps.bzl", "buildifier_depen
 
 buildifier_dependencies()
 
+# We don't use any nodejs but this includes a rule for publishing releases to npm
```
[the remark this answers is not in the document]
Comment (sudoforge) [reply WORKSPACE]: Correct, the `_maybe()` call for `rules_nodejs` should be removed from `buildifier_dependencies()`

**Comment by alexeagle** [comment]
By the way @vladmos when you cut the next release, please let me or @soldair know if you have any problems with the npm publish step - after that I'll work on making format and lint actions enabled by default in new projects created with `npm init @bazel`

File: WORKSPACE (hunk 1 above)
[the remark this answers is not in the document]
Comment (vladmos) [reply WORKSPACE]: Sure, I removed it from `deps.bzl` initially, but then accidentally reverted the change.

**Pull request merged**